assets/src/editor/hocs/editor-modal/with-editor.js

```diff
@@ -13,27 +13,45 @@ const { confirm } = window;
  * wraps the component that contains the withEditorModal
  *
  * @function
+ * @param {string} closeEditorNotice 	message displayed if user attempts
+ * 										to close modal when changes are not
+ * 										yet saved. To override the appearance
+ * 										of the closeEditorNotice, simply pass
+ * 										an empty string for this prop
+ * @param {function} onOpen    			fires when modal opens
+ * @param {function} onClose 			fires when modal closes
```

Comment:
Hmm, I'm not sure `onClose` or `onOpen` needs to be handled in this component.  `withEditorModal` already accepts them (and handles them) - I think that's where they belong. If there's issues with things not working the way you expect using that logic then that needs investigated but not in this pull. There also was a card created for me in the project board to evaluate this and see if it was necessary - is that card still needed - or is this something you're going to take care of after all?

Reply:
this was something you had suggested in your first review for this pull !

from: https://github.com/eventespresso/event-espresso-core/pull/1265#discussion_r289376219

> This will affect submit-less editor modals as changes aren't "saved" via a submit but update in the state via field listeners. I think you started in the right direction by checking for whether closeEditorNotice is an empty string in the toggleEditor function, but on construct it sets a default if it's falsey. So some options for allowing for submit-less components are:
> 
> - only set a default for closeEditorNotice if the prop is undefined (rather than falsey).
> - or, accept a showNoticeOnClose boolean prop that controls whether the notice logic executes on toggling the editor closed.
> - or, accept an optional onClose callback prop that defaults to the current logic you added here, but can be overridden by parent components if implementations don't want that behaviour (or to enhance with any other behaviour).
> 
> I kind of prefer the last option myself (you could even accept onOpen, and onClose callback props for enhancing the toggle).

Reply:
I did, but to emphasize, there _was a card created for me in the project list to examine_. to which I added an additional note after the card was created. https://www.evernote.com/l/AAN4rI6OOl1Jeb8TSlN9sfAloLWt-1_Aot8B/image.png

In that pull comment you also missed out this line:

> I'm fine with having any of the above done in a separate pull - and I don't mind doing (but there should be a card created for it).

You decided to go ahead and fix this in this pull which in turn caused me to take a closer look at the other components and lead to my additional feedback in this review.